lib/gltfPipeline.js

```diff
@@ -26,13 +29,24 @@ module.exports = {
     processFileToDisk : processFileToDisk
 };
 
-function processJSON(gltf, options, callback) {
+function writeSources(gltf, outputPath, options, callback) {
+    var embed = options.embed;
+    var embedImage = options.embedImage;
+    var basePath = path.dirname(outputPath);
+    async.each(['buffers', 'images', 'shaders'], function(name) {
+        writeSource(gltf, basePath, name, embed, embedImage, callback);
+    });
+}
+
+function processJSON(gltf, outputPath, options, callback) {
     addPipelineExtras(gltf);
     loadGltfUris(gltf, options, function(err, gltf) {
         if (err) {
             throw err;
         }
-        processJSONWithExtras(gltf, options, callback);
+        processJSONWithExtras(gltf, options, function(gltf) {
+            writeSources(gltf, outputPath, options, callback(gltf));
```

Comment:
What will happen is the result of the `callback(gltf)` will be sent to `writeSources`, which could be anything and is probably not itself a callback.